src/Temporalio/Client/TemporalConnectionOptions.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Options for <see cref="TemporalConnection.ConnectAsync" />.
+    /// <see cref="TemporalConnectionOptions.TargetHost" /> is required.
+    /// </summary>
+    public class TemporalConnectionOptions : ICloneable
+    {
+        /// <summary>
+        /// Create default options.
+        /// </summary>
+        public TemporalConnectionOptions() { }
+
+        /// <summary>
+        /// Create default options with a target host.
+        /// </summary>
+        /// <param name="targetHost">Target host to connect to.</param>
+        /// <seealso cref="TemporalConnectionOptions.TargetHost" />
+        public TemporalConnectionOptions(string targetHost) : this()
+        {
+            this.TargetHost = targetHost;
+        }
+
+        /// <summary>
+        /// Gets or sets the Temporal server <c>host:port</c> to connect to.
+        /// </summary>
+        /// <remarks>
+        /// This is required for all connections.
+        /// </remarks>
+        public string? TargetHost { get; set; }
+
+        /// <summary>
+        /// Gets or sets the TLS options for connection.
+        /// </summary>
+        /// <remarks>
+        /// This must be set, even to a default instance, to do any TLS connection.
+        /// </remarks>
+        public TlsOptions? TlsOptions { get; set; }
+
+        /// <summary>
+        /// Gets or sets retry options for this connection.
+        /// </summary>
+        /// <remarks>
+        /// This only applies if the call is being retried, which by default is usually all
+        /// high-level client calls but no raw gRPC calls.
+        /// </remarks>
+        public RpcRetryOptions? RpcRetryOptions { get; set; }
```

Comment:
We do in Python and I think it's reasonable here. If we used the lang-native gRPC in either of those, I'd tell the user to go use the retry config for that, but since both of those SDKs use ours, this gives the flexibility needed for their retry needs on raw gRPC calls.